DQMServices/Core/interface/DQMEDAnalyzer.h

```diff
@@ -1,80 +1,54 @@
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/LuminosityBlock.h"
+#include "FWCore/Framework/interface/Run.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+#include "FWCore/ServiceRegistry/interface/ModuleCallingContext.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+
+class DQMEDAnalyzer : public edm::one::EDAnalyzer<edm::one::WatchRuns, edm::one::WatchLuminosityBlocks>
 {
 public:
-  DQMEDAnalyzer();
-  // implicit copy constructor
-  // implicit assignment operator
-  // implicit destructor
-  void beginStream(edm::StreamID id) final;
-  void beginRun(edm::Run const &, edm::EventSetup const&) final;
-  static std::shared_ptr<dqmDetails::NoCache> globalBeginRunSummary(edm::Run const&,
-                                                        edm::EventSetup const&,
-                                                        RunContext const*);
-  void endRunSummary(edm::Run const&,
-                             edm::EventSetup const&,
-                             dqmDetails::NoCache*) const final;
-  static void globalEndRunSummary(edm::Run const&,
-                                  edm::EventSetup const&,
-                                  RunContext const*,
-                                  dqmDetails::NoCache*);
-  static std::shared_ptr<dqmDetails::NoCache> globalBeginLuminosityBlockSummary(edm::LuminosityBlock const&,
-                                                                    edm::EventSetup const&,
-                                                                    LuminosityBlockContext const*);
-  void endLuminosityBlockSummary(edm::LuminosityBlock const&,
-                                         edm::EventSetup const&,
-                                         dqmDetails::NoCache*) const final;
-  static void globalEndLuminosityBlockSummary(edm::LuminosityBlock const&,
-                                              edm::EventSetup const&,
-                                              LuminosityBlockContext const*,
-                                              dqmDetails::NoCache*);
-  uint32_t streamId() const {return stream_id_;}
+  DQMEDAnalyzer() = default;
+  ~DQMEDAnalyzer() = default;
+  DQMEDAnalyzer(DQMEDAnalyzer const&) = delete;
+  DQMEDAnalyzer(DQMEDAnalyzer &&) = delete;
+
+  void beginRun(edm::Run const& run, edm::EventSetup const& setup) final
+  {
+    dqmBeginRun(run, setup);
+    edm::Service<DQMStore>()->bookTransaction(
+      [this, &run, &setup](DQMStore::IBooker & booker)
+      {
+        booker.cd();
+        this->bookHistograms(booker, run, setup);
+      },
+      run.run(),
+      run.moduleCallingContext()->moduleDescription()->id());
+  }
+
+  void endRun(edm::Run const& run, edm::EventSetup const& setup) override
+  { }
+
+  void beginLuminosityBlock(edm::LuminosityBlock const& lumi, edm::EventSetup const& setup) override
```

Comment:
can this be final?

Reply:
Feel free to make a PR.

Reply:
At least one module overrides this with non-trivial logic, so no, not for now.